### PR TITLE
docs: add contribution guidelines

### PR DESCRIPTION
**Pull request by matthiasschaub**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Run flake8 and Pylint with following commands:
+
+```bash
+flake8 .
+pylint .
+```
+
+*Tips:*
+- Mark in-line that flake8 should not raise any error: `print()  # noqa`
```
Comment (mhhd2020) [inline CONTRIBUTING.md:59]: We might also add   
`# pylint: disable=rule-name`  
and perhaps (if we included Bandit above)  
`# nosec`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### HTML
+
+[.htmlhintrc](.htmlhintrc) and the [documentation of these rules](https://github.com/htmlhint/HTMLHint/tree/master/docs/user-guide/rules)
```
Comment (mhhd2020) [inline CONTRIBUTING.md:63]: There's a "See" missing, right?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+[stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard) with some modifications (see [.stylelintrc.json](.stylelintrc.json))  
+
+## Commit Hooks
+
+If you discover any violations in the existing code, feel very welcome to fix them. To facilitate paying attention to these conventions, please make sure that for your pull request all checks succeed. You can also set up git hooks to automatically run relevant linters before your code is committed, see [.hooks/README.MD](.hooks/README.MD) for more info. Please also make sure that everything you modify or add is covered by [unit tests](test).
+
+## Tests
+
+Please provide tests.
```
Comment (mhhd2020) [inline CONTRIBUTING.md:79]: I think we should specify which frameworks we use for the tests and also which kinds of tests, e.g. unit tests with pytest. System tests are not done automatically yet, but we could already think of how it should be done in the future

**Review by mhhd2020 — COMMENTED** [review]
Thank you for this pull request, good to have a separate CONTRIBUTING.md!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+Each PR has to be approved by at least one reviewer before merging it. A person can be assigned as reviewer by either mark them as such or asking for a review by tagging the person in the description/comment of the PR.
+
+A draft pull request can be made even if the branch is not ready to be merged yet. This way people/reviewer know that there is someone still working on that branch actively. This gives them the opportunity share their thoughts knowing that the pull request is still subject to change and does not need a full review yet.
+
+The [CHANGELOG.md](CHANGELOG.md) describes changes made in a pull request. It should contain a short description of the performed changes, as well as (a) link(s) to issue(s) or pull request.
+
+#### Review Process
+
+1. Dev makes a PR.
+2. Rev reviews and raises some comments.
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply CONTRIBUTING.md:28]: Done

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+All development work is based on the main branch (`main`). Pull requests are expected to target the `main` branch.
+
+## Style Guide
+
+### Python
+
+In short:
+- Guideline: [PEP8](https://peps.python.org/pep-0008/), [PEP 484 (Type Hints)](https://peps.python.org/pep-0484/)
+- Linter: [flake8](https://flake8.pycqa.org), [Pylint](https://pylint.pycqa.org/)
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply CONTRIBUTING.md:45]: Dane

File: CONTRIBUTING.md (hunk 1 above)
Comment (matthiasschaub) [reply CONTRIBUTING.md:59]: Done

File: CONTRIBUTING.md (hunk 3 above)
Comment (matthiasschaub) [reply CONTRIBUTING.md:79]: Okay. What do you suggest here? Just ask for unit tests using the Pytest framework?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+#### Linters
+
+This project uses [flake8](https://flake8.pycqa.org), [Pylint](https://pylint.pycqa.org/), [mypy](http://mypy-lang.org/) and [bandit](https://github.com/PyCQA/bandit) to ensure consistent code style.
+
+Run those linter's with following commands:
+
+```bash
+flake8 .
+pylint .
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply CONTRIBUTING.md:56]: Done

**Review by mhhd2020 — APPROVED** [review]

**Review by mhhd2020 — APPROVED** [review]

**Pull request merged**